lib/charms/opensearch/v0/helper_cluster.py

```diff
@@ -57,6 +57,27 @@ def suggest_roles(nodes: List[Node], planned_units: int) -> List[str]:
 
         return base_roles + ["cluster_manager"]
 
+    @staticmethod
+    def get_cluster_settings(
+        opensearch: OpenSearchDistribution,
+        host: Optional[str] = None,
+        alt_hosts: Optional[List[str]] = None,
+        include_defaults: bool = False,
+    ) -> Dict[str, any]:
+        """Get the cluster settings."""
+        settings = opensearch.request(
+            "GET",
+            f"/_cluster/settings?flat_settings=true&include_defaults={str(include_defaults).lower()}",
+            host=host,
+            alt_hosts=alt_hosts,
+        )
+
+        result = {}
+        for el in ["defaults", "transient", "persistent"]:
```

Comment:
is this needed? the output seems to be exactly those 3 keys

Reply:
I am not sure this is always the case, tbh. There is nothing in the docs that gives me that guarantee

Reply:
Okay, I will set to return only the "transient" value, as this is what the docs in ES states it overrides everything else: https://www.elastic.co/guide/en/elasticsearch/reference/current/cluster-update-settings.html

Reply:
Hey @Mehdi-Bendriss, no luck :(

After this change, I can see our CI failed, and there was a reboot early on:
```
DEBUG unit.opensearch/1.juju-log server.go:325 opensearch-peers:0: Difference between current and new configuration: 
{'in current but not in new_conf': {}, 'in new_conf but not in current': {'knn.plugin.enabled': 'true'}, 'in both but different values': {}}
DEBUG unit.opensearch/1.juju-log server.go:325 opensearch-peers:0: Status: installed
```

Readding the original for (on a simplified structure, but still the same basic logic), resolves the issue: https://pastebin.ubuntu.com/p/8R3JJmVgV3/